### PR TITLE
planning scene ros api

### PR DESCRIPTION
**Pull request by mlautman**
[Live version](http://tutorials.picknik.ai/doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.html)

### CODE REVIEW

File: _static/override.css [hunk 1]
```diff
@@ -20,3 +20,6 @@
   text-align: right;
   margin-bottom: -25px;
 }
+.red {
+  color: red;
```
Comment (davetcoleman) [inline _static/override.css]: where is this used? for the code string color?

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 2]
```diff
@@ -8,27 +8,26 @@ two operations:
...
+-------------
 If you haven't already done so, make sure you've completed the steps in `Prerequisites
 <../prerequisites/prerequisites.html>`_.
 
-.. tutorial-formatter:: ./src/planning_scene_ros_api_tutorial.cpp
 
 The entire code
-^^^^^^^^^^^^^^^
+---------------
 The entire code can be seen :codedir:`here in the MoveIt! github project<planning_scene_ros_api>`.
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: "GitHub"

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 3]
```diff
@@ -8,27 +8,26 @@ two operations:
...
 
-Compiling the code
-^^^^^^^^^^^^^^^^^^
-Follow the `instructions for compiling code from source <http://moveit.ros.org/install/>`_.
+.. tutorial-formatter:: ./src/planning_scene_ros_api_tutorial.cpp
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: I prefer to show *then* tell... let the user run the roslaunch and see the actions, then read about what happened to better understand. Reading is boring unless you first peak their curiosity. lets not flip the ordering of the tutorial.

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 4]
```diff
@@ -8,27 +8,26 @@ two operations:
...
 
 The launch file
-^^^^^^^^^^^^^^^
-The entire launch file is :codedir:`here <planning_scene_ros_api/launch/planning_scene_ros_api_tutorial.launch>` on github. All the code in this tutorial can be compiled and run from the moveit_tutorials package
-that you have as part of your MoveIt! setup.
+---------------
+The entire launch file is :codedir:`here <planning_scene_ros_api/launch/planning_scene_ros_api_tutorial.launch>` on github. All the code in this tutorial can be compiled and run from the moveit_tutorials package.
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: please do a findreplace github GitHub for the tutorials in a different PR

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 5]
```diff
@@ -42,3 +41,7 @@ In rviz, you should be able to see the following:
  * Object gets attached to the robot
  * Object gets detached from the robot
  * Object is removed from the planning scene
+
+.. role:: red
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: does not work... this section is not red

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 6]
```diff
@@ -42,3 +41,7 @@ In rviz, you should be able to see the following:
...
+
+**Note:** You may see an error message reading :red:`Found empty JointState message`. This is a known cosmetic bug and likely related to `Issue #579 <https://github.com/ros-planning/moveit/issues/579>`_
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: is "cosmetic bug" really the word for this? its just a bug...

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp [hunk 7]
```diff
@@ -55,8 +55,7 @@ int main(int argc, char **argv)
   spinner.start();
 
   ros::NodeHandle node_handle;
-  ros::Duration sleep_time(10.0);
-  sleep_time.sleep();
+  ros::Duration sleep_time(2.0);
   sleep_time.sleep();
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp]: can you replace all the sleep timers with visual_tools_->next() functionality like in the move group interface tutorial?

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp [hunk 8]
```diff
@@ -159,19 +160,20 @@ int main(int argc, char **argv)
...
   /* Carry out the REMOVE + ATTACH operation */
-  ROS_INFO("Attaching the object to the right wrist and removing it from the world.");
+  ROS_INFO("Attaching the object to the wrist and removing it from the world.");
   planning_scene.world.collision_objects.clear();
   planning_scene.world.collision_objects.push_back(remove_object);
   planning_scene.robot_state.attached_collision_objects.push_back(attached_object);
   planning_scene_diff_publisher.publish(planning_scene);
 
+  ROS_INFO("Sleeping for 2s to display changes.");
   sleep_time.sleep();
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp]: remove

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp [hunk 9]
```diff
@@ -199,6 +201,7 @@ int main(int argc, char **argv)
   planning_scene.is_diff = true;
   planning_scene_diff_publisher.publish(planning_scene);
 
+  ROS_INFO("Sleeping for 2s to display changes.");
   sleep_time.sleep();
```
Comment (davetcoleman) [inline doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp]: i hate sleeps

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst (hunk 5 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: It works on my local build. I've been overwriting the folders on the server and I missed overwriting the CSS.

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst (hunk 6 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: Yup, I didn't want new users freaking out about it so I borrowed language from one of the GitHub issues. I'll change it to known bug.

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp (hunk 7 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp]: Totally in agreement here. I was planning on adding visual_tools to all of the tutorials with planners. I think the plot functionality will add a lot to the demos.

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp (hunk 8 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp]: Will be replaced with visual_tools_->prompt("press 'next' .... ")

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp (hunk 9 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp]: me too

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst (hunk 3 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: I'm totally for this. I'm inclined to leave this as is for consistency across the tutorials for now and then fix all of them together in a single PR

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst (hunk 5 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: It should work now

File: _static/override.css (hunk 1 above)
Comment (mlautman) [reply _static/override.css]: yup

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst (hunk 3 above)
Comment (mlautman) [reply doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst]: I fixed the order on this pr and added *reorder to show then tell* to my notes 
https://docs.google.com/document/d/1r7FVPEzharYeF4xblLIb37zJRUtllhPBeP9-rQ7tzBQ/edit

**Comment by mlautman** [comment]
@davetcoleman Ready for merge?

**Comment by mlautman** [comment]
@davetcoleman added mvt and some other minor fixes. Ready for another review

**Pull request merged**